main/src/ca/uwaterloo/flix/runtime/Benchmark/FlixListInScala.scala

```diff
@@ -0,0 +1,201 @@
+/*
+ *  Copyright 2017 Ramin Zarifi
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package ca.uwaterloo.flix.runtime.Benchmark
+
+// Implementation of Flix List in Scala
+object FlixListInScala {
+  // This trait represents `List` enum in Flix.
+  sealed trait FlixList[+T]
+
+  // This object represents `Nil` case in `List`
+  case object FlixNil extends FlixList[Nothing]
+
+  // This case class represents `Cons` case in `List`
+  case class FlixCons[+T](field: (T, FlixList[T])) extends FlixList[T]
```

Comment:
Ok, this is definitely apples-to-apples. But for it to be idiomatic scala, we have to change it to:

```
case class FlixCons(x: T, rs: FlixList[T])
```

Otherwise people will argue that we are giving ourselves an unfair advantage. 

If you are wondering what we want to compare ourselves to, it should be this: 

"Idiomatic, but purely functional, Scala code".